### ISSUE TITLE
[REQUEST] DCA_max_pairs

### ISSUE THREAD

**Issue by Q-Rush** [taniman/profit-trailer#554]
Hi,

since it can happen that all of your pairs slip into the DCA Log it can happen that your balance is not big enough to buy in for all pairs. 

I would request a setting like **DCA_max_pairs** or **DCA_limit_pairs** .

For example i got 9 DCA pairs. The highest one has a profit of -4% and the lowest -15%.
My Balance is too low to buy in 5 doubles for each. But my balance would be high enough to buy in 5 doubles for the first pair. The profit could allow DCA to buy in 5 doubles for the next pair and so on.

This would allow people to use more of their balance and worry less about bags.

**Comment by Q-Rush** [comment]
#####

**Comment by nstefi** [comment]
Good idea.

I also thought that it would be nice to have some manual control in the DCA Log screen. We could have some "Pause" buttons for example on certain DCA pairs there, so it would not buy more of that coin if the price drops lower. Then lets's say when the rest of the DCA pairs that you allow would sell out, your balance would grow and you could release the paused ones too. This would be faster as a temporary solution than changing parameters.
But having DCA_max_pairs or DCA_limit_pairs can be a long term solution to set them as global parameters.

**Comment by taniman** [comment]
I do not see the added benefit, since you can enable your DCA items 1 by 1 and thus there is no need then for a DCA_max_pairs

**Comment by nstefi** [comment]
Where can you enable the your DCA items 1 by 1?

**Comment by Q-Rush** [comment]
@taniman i dont see an option to enable DCA one by one. At least the wiki states nothing about such a function. :(

**Comment by taniman** [comment]
instead of using ALL_DCA_enabled = true

you can do.

ALL_DCA_enabled = false
BTC-ETH_DCA_enabled = true (this is a bittrex example)

On the wiki you can see what properties starting with ALL_ can be converted to a coinpair property too.

**Comment by nstefi** [comment]
I see. so this is in the PAIRS file, not the DCA file.
If I make set false for a pair that's already in the DCA, can it be resumed later, after let's say my other DCA pairs sell out?

**Comment by taniman** [comment]
yes, that is also possible

**Comment by nstefi** [comment]
So I guess my idea of a "Pause" button in the interface works the same way as setting those parameters (BTC-ETH_DCA_enabled = false)
Maybe in the future it would be nice to have a button in the interface doing the same thing

**Comment by Q-Rush** [comment]
@taniman that is right. but the benfit of having dca_max_pairs would be that this would happen automatically. you would only set it to a specific amount of pairs and when any combination appears it triggers to focus on the first appeared pair. 
your solution would mean that i would have to babysit the bot and wait for each pair to be finished in dca to change it to the next one.
i hope i descibe it correctly so you can see the benefit

**Comment by nstefi** [comment]
@Q-Rush if you set a dca_max_pairs parameter, then a new pair wouldn't get a chance to enter in DCA, even though it may have a quicker chance to solve and get out. Would it be better if the bot could decide based on the list of DCA which ones have more loss (higher negative profit), and pause those until you have enough balance? I am not sure if it's worth to automate that way instead.
So I'm thinking to allow them in DCA but decide later to pause the more riskier ones.
Maybe based on the total cost, not profit?

**Comment by taniman** [comment]
I understand your point. But I am sure that you want also that those coinpairs stay in pairs log if the max is reached. 

That has many complications. Because currently a pair is DCA enablled or not. If we have to the the max.
We will have to keep a list or all items in DCA at the moment. And make sure the other items go to PAIS.

All in all not as simple as it sounds

I can reopen this. But it will stay will low priority

**Comment by Q-Rush** [comment]
Alright thank you! In the meantime i can try to write a script that allows me to that :)

**Comment by bluefuzz** [comment]
May I add my vote for this feature. DCA requires far too much babysitting if we are unable able to specify the max number of pairs at a time. DCA_max_pairs is probably my single 'most wanted' feature ...

**Comment by JonTea** [comment]
> That has many complications. Because currently a pair is DCA enablled or not. If we have to the the max. We will have to keep a list or all items in DCA at the moment. And make sure the other items go to PAIS. All in all not as simple as it sounds

The way I imagined it would be to set a DCA limit at each trigger level.
e.g.
buy_trigger_1_max_trading_pairs = 6
buy_trigger_2_max_trading_pairs = 4
buy_trigger_3_max_trading_pairs = 1
buy_trigger_4_max_trading_pairs = 1

That way they'd still show in the DCA list, but it wouldn't buy more once it hits the limit.  I think that would help get rid of the stubborn bags without having to configure each coin separately.  Could that reduce the complexity you mentioned @taniman ?  I guess the behaviour would be the same as when max_cost is exceeded in DCA.properties

**Comment by nstefi** [comment]
I guess the "DCA_max_pairs" should be defined in the PAIRS.properties file, and somewhere in the code where the bot checks for this parameter ALL_DCA_enabled (in my case "-5"), and checks if the profit(loss) is less than -5% for a specific pair, it should also check how many pairs are in the DCA log at the moment, and if there are >= than DCA_max_pairs (I guess it would never be >, but include that for safety), then it would leave the pair in the Pairs Log. otherwise move it to DCA Log, and apply DCA algorithm from there onward.
> "We will have to keep a list or all items in DCA at the moment. And make sure the other items go to PAIRS"
I guess you need to keep a list of items in the DCA, because they enter in DCA, you should not take them out from there unless you modify the config to exclude a specific pair (BTC-ETH_DCA_enabled = false).
Or if you want to allow to modify the DCA_max_pairs in config (to a lower number), I am not sure which pair you should remove from DCA to Pairs.

**Comment by ACETyr** [comment]
I think it would be more efficient to serialize DCA in a DCA_max_concurrent = x style of setting. Instead of not putting more than X pairs in DCA, you would basically only let x amount of pairs buying while all others in DCA are put on SOM. The problem is not too many in DCA in general but the lack of funds to DCA multiple in parallel. So some serialisation of DCA might help in restricted budget situations.

**Comment by hhaick** [comment]
It would be nice if we could set DCA values individually for diferent pairs. I mean, we could choose between a set of dca configurarions for all pairs or a set of different dca configurations for each individual pair.
E.g.:
ETCBTC_buy_trigger1= 2
ETCBTC_buy_trigger2= 3
XRPBTC_buy_trigger1 = 1
XRPBTC_buy_trigger2 = 7
XRPBTC_buy_trigger3 = 18

**Comment by RunPlz** [comment]
Yes definitely this would enchance the BOT and new / better strategies could arise.
Selecting each by one is no solution in comparison to what is being post here.

**Comment by angelsomar** [comment]
The truth is that the management of the DCA is linked to the management of our risk capital, and you can already set priorities in DCA to invest only in those most likely to leave soon

example
do not buy if the volume decreases too much
DEFAULT_DCA_min_buy_volume = 500

and in buying strategies you can say only buy if I detect an uptrend, so if the pair goes into a tailspin, do not invest

Finally, as a simple strategy, I tell you to manage your strategy as if you only have 50% of your capital, success

**Comment by cvthub** [comment]
The feature will be available in 2.1.
Using DCA_max_trading_pairs the amount of coins which are DCA-ed can be limited.
